src/uid.js

```diff
@@ -1,14 +1,27 @@
-let count = 0;
-
-export default function(name) {
-  return new Id("O-" + (name == null ? "" : name + "-") + ++count);
-}
-
-function Id(id) {
-  this.id = id;
-  this.href = window.location.href + "#" + id;
-}
-
-Id.prototype.toString = function() {
-  return "url(" + this.href + ")";
-};
+let count = 0;
+
+class Id {
+  constructor(id) {
+    this._id = id;
+  }
+
+  get id() {
+    return this._id;
+  }
+
+  get href() {
+    const url = new URL(window.location);
+    url.hash = this.id;
+    return url.href;
+  }
+
+  toString() {
+    return `url(${this.href})`;
+  }
+}
```

Comment:
Previous version of id has cached version of href.

Reply:
True that, failed to figure out that is actually important :upside_down_face: I'll cache it again.